start_admin_sshd.sh

```diff
@@ -8,36 +8,87 @@ mkdir -p /home/ec2-user/.ssh/
 chmod 700 /home/ec2-user/.ssh/
 ssh_host_key_dir="/.bottlerocket/host-containers/admin/etc/ssh"
 ssh_config_dir="/home/ec2-user/.ssh"
+sshd_config="/etc/ssh/sshd_config"
+user_data="/.bottlerocket/host-containers/admin/user-data"
+available_auth_methods=0
 
-# Populate authorized_keys with all the public keys found in instance meta-data
-# The URLs for keys include an index and the keypair name, e.g.
-# http://169.254.169.254/latest/meta-data/public-keys/0=mykeypair/openssh-key
-ssh_authorized_keys="${ssh_config_dir}/authorized_keys"
-touch ${ssh_authorized_keys}
-chmod 600 ${ssh_authorized_keys}
-public_key_base_url="http://169.254.169.254/latest/meta-data/public-keys/"
-imds_session_token=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 60")
-imds_request_add_header="X-aws-ec2-metadata-token: ${imds_session_token}"
-public_key_indexes=($(curl -H "${imds_request_add_header}" -sf "${public_key_base_url}" \
-    | cut -d= -f1 \
-    | xargs))
-
-for public_key_index in "${public_key_indexes[@]}"; do
-  public_key_data="$(curl -H "${imds_request_add_header}" -sf "${public_key_base_url}/${public_key_index}/openssh-key")"
-  if [[ ! "${public_key_data}" =~ ^"ssh" ]]; then
-    echo "Key ${public_key_data} with index ${public_key_index} looks invalid" >&2
-    continue
+set -f
+IFS='
+'
+
+# Is user-data does not contain a JSON block, there's not much point in continuing.
+if ! jq empty ${user_data} &>/dev/null; then
+    echo "${user_data} does not contain valid JSON." >&2
+fi
+
+# Populate authorized_keys with all the public keys found in api user-data
+if grep -q authorized_keys ${user_data}; then
+  ssh_authorized_keys="${ssh_config_dir}/authorized_keys"
+  touch ${ssh_authorized_keys}
+  chmod 600 ${ssh_authorized_keys}
```

Comment:
There's mixed use of `${varname}`, `"${varname}"` and `"$varname"` - can you update this to use one of the quoted variants? I'm partial to `"$varname"` where possible, but am okay with `"${varname}"` too.

```suggestion
if grep -q authorized_keys "$user_data"; then
  ssh_authorized_keys="$ssh_config_dir/authorized_keys"
  touch "$ssh_authorized_keys"
  chmod 600 "$ssh_authorized_keys"
```